Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -790,43 +790,60 @@ namespace OpenInfraPlatform {
 					}
 				}
 				
+				/*! \brief  Converts a list of \c IfcFaceBound -s to a list of boundary loops.
+				 * The first loop in the return list is converted from the \c IfcFaceOuterBound entity.
+				 * Only one boundary should be of this type.
+				 *
+				 * \param[in]	ifcFaceBounds	List of \c IfcFaceBound entities to be converted.
+				 * \param[in]	pos				The relative location of the origin of the representation's coordinate system within the geometric context.
+				 * \return		List of boundary loops.
+				 */
 				std::vector<std::vector<carve::geom::vector<3>>> convertIfcFaceBoundList(
-					const std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>>& ifcFaceBounds,
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>> ifcFaceBounds,
 					const carve::math::Matrix& pos) const noexcept(true)
 				{
-					const std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>> faceBounds = swapOuterBoundaryToFront(ifcFaceBounds);
+					swapOuterBoundaryToFront(ifcFaceBounds);
 
 					std::vector<std::vector<carve::geom::vector<3>>> faceBoundLoops;
-					faceBoundLoops.resize(faceBounds.size());
-					for (size_t i = 0; i < faceBounds.size(); i++)
-					{
-						// get vertices of one bound loop into faceBoundLoops[i]
-						faceBoundLoops[i] = convertIfcFaceBound(faceBounds[i], pos);
-					}
+					faceBoundLoops.resize(ifcFaceBounds.size());

```

Comment:
This can done in a single line, to reserve the amount of space in a vector, I think it is this syntax:

`std::vector<std::vector<carve::geom::vector<3>>> faceBoundLoops(ifcFaceBounds.size());`

Reply:
Thanks for this hint! I had forgot about that.
Done in commit 3f00fd6.